### PR TITLE
update to sdk-v2

### LINKED ISSUE #46: Migrate to aws-sdk-go v2
**What would you like to be added**:
Migrate usage of v1 aws-go-sdk to v2.

Similar to https://github.com/gardener/gardener-extension-provider-aws/issues/1054

**Why is this needed**:
[aws-sdk-go v1](https://github.com/aws/aws-sdk-go) is in maintenance mode. EOF will be reached on 31.7.2025. In the context of https://github.com/gardener/gardener/issues/9586 it would be great if we can directly implement the new functionalities with the [aws-sdk-go v2](https://github.com/aws/aws-sdk-go-v2) which provides a nicer API for working with JWTs https://github.com/aws/aws-sdk-go-v2/blob/f1d71c59a1499981873f70db8c92d07242779670/credentials/stscreds/web_identity_provider.go#L70

### PR DESCRIPTION
**Pull request by kon-angelo**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #46 

**Special notes for your reviewer**:

**Release note**:

```feature user
Update sdk version to v2
```

### CODE REVIEW

**Comment by kon-angelo** [comment]
cc: @AndreasBurger @MartinWeindel

File: go.mod [hunk 1]
```diff
@@ -4,6 +4,10 @@ go 1.23.0
 
 require (
 	github.com/aws/aws-sdk-go v1.55.5
```
Comment (MartinWeindel) [inline go.mod:6]: Not needed anymore.
```suggestion
```

**Review by MartinWeindel — APPROVED** [review]
/lgtm

File: pkg/updater/ec2.go [hunk 2]
```diff
@@ -28,50 +32,50 @@ const TagNameKubernetesClusterLegacy = "KubernetesCluster"
...
 	}
 	return false
 }
 
-func getNameTagValue(tags []*ec2.Tag) string {
+func getNameTagValue(tags []ec2types.Tag) string {
 	for _, tag := range tags {
-		if aws.StringValue(tag.Key) == "Name" {
-			return aws.StringValue(tag.Value)
+		if ptr.Deref(tag.Key, "") == "Name" {
```
Comment (AndreasBurger) [inline pkg/updater/ec2.go:77]: You could also use `aws.ToString()`, up to you.

**Review by AndreasBurger — APPROVED** [review]
lgtm

**Pull request merged**